python/GafferSceneUI/HistoryPathDialogue.py

```diff
@@ -0,0 +1,193 @@
+
+		data.icon = {
+			Gaffer.TweakPlug.Mode.Replace : "replaceSmall.png",
+			Gaffer.TweakPlug.Mode.Add : "plusSmall.png",
+			Gaffer.TweakPlug.Mode.Subtract : "minusSmall.png",
+			Gaffer.TweakPlug.Mode.Multiply : "multiplySmall.png",
+			Gaffer.TweakPlug.Mode.Remove : "removeSmall.png",
+			Gaffer.TweakPlug.Mode.Create : "createSmall.png",
+			Gaffer.TweakPlug.Mode.Min : "lessThanSmall.png",
+			Gaffer.TweakPlug.Mode.Max : "greaterThanSmall.png",
+			Gaffer.TweakPlug.Mode.ListAppend : "listAppendSmall.png",
+			Gaffer.TweakPlug.Mode.ListPrepend : "listPrependSmall.png",
+			Gaffer.TweakPlug.Mode.ListRemove : "listRemoveSmall.png",
+		}.get( cellValue, "errorSmall.png" )
+
+		return data
+
+	def headerData( self, canceller = None ) :
+
+		return self.CellData( self.__title )
+
+class HistoryPathDialogue( GafferUI.Dialogue ) :
+
+	def __init__( self, path, scriptNode, title=None, **kw ) :
+
+		assert( isinstance( path, GafferSceneUI.Private.Inspector.HistoryPath ) )
+		assert( isinstance( scriptNode, Gaffer.ScriptNode ) )
+
+		if title is None :
+			title = "History"
+
+		GafferUI.Dialogue.__init__( self, title, **kw )
+
+		self.__scriptNode = scriptNode
+
+		self.__path = path
+		self.__pathListingWidget = GafferUI.PathListingWidget(
+			Gaffer.DictPath( {}, "/" ),
+			columns = (
+				GafferUI.PathListingWidget.defaultNameColumn,
+				GafferUI.PathListingWidget.StandardColumn( "Value", "history:value" ),
+				_OperationIconColumn( "Operation", "history:operation" ),
+			),
+			sortable = False,
+			horizontalScrollMode = GafferUI.ScrollMode.Automatic,
+			selectionMode = GafferUI.PathListingWidget.SelectionMode.Cell
+		)
+		self.__pathListingWidget.setDragPointer( "values" )
+
+		self.__nameColumnIndex = 0
+		self.__valueColumnIndex = 1
+		self.__operationColumnIndex = 2
+
+		self.__pathListingWidget.buttonDoubleClickSignal().connectFront( Gaffer.WeakMethod( self.__buttonDoubleClick ), scoped = False )
+		self.__pathListingWidget.keyPressSignal().connectFront( Gaffer.WeakMethod( self.__keyPress ), scoped = False )
+		self.__pathListingWidget.dragBeginSignal().connectFront( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+		self._setWidget( self.__pathListingWidget )
+
+		self.__pathListingWidget.setPath( self.__path )
+
+	def __buttonDoubleClick( self, pathListing, event ) :
+
+		if pathListing.pathAt( event.line.p0 ) is None :
+			return False
+
+		if event.buttons == event.Buttons.Left :
+			self.__editSelectedCell( pathListing )
+
+			return True
+
+		return False
+
+	def __keyPress( self, pathListing, event ) :
+
+		if event.key == "Return" or event.key == "Enter" :
+			self.__editSelectedCell( pathListing )
+
+			return True
+
+		return False
+
+	def __editSelectedCell( self, pathListing ) :
+
+		selection = pathListing.getSelection()
+
+		selectedPath, selectedColumn = self.__selectionData( selection )
+
+		if selectedColumn == self.__nameColumnIndex :
+			editPlug = selectedPath.property( "history:acquiredEdit" )
```

Comment:
Unused

Reply:
Fixed in 3a3e0a61ec85c003d82b1873be8eef4f7d5ff6f0